Optimised shell code

doasu opened the pull request:
Replaced `TMUX_UID`,`SERVER`,`DATA` with two variables:

> 1. **SOCKET_PATH ::** `#{socket_path}` **::** expands to `/tmp/tmux-1000/default` or `/tmp/tmux-$TMUX_UID/$SERVER`
> 2. **DATA ::** `${SOCKET_PATH#/tmp/tmux-}` **::** this expands the previous variable to `/tmp/tmux-keylocker`+`1000/default` or +`$TMUX_UID/$SERVER`

Removed superfluous commands from the `TK_HOME` variable:
> AFAIK, there's nothing wrong with using just `$(dirname)` for this.
> This is how `tmux display-message "$TK_HOME"` looks after the change:
> ![image](https://github.com/TheSast/tmux-keylocker/assets/99696763/88f561e9-641c-456c-8489-de4411161786)

doasu committed:
toggle.sh :: optimised shell code

replaced multiple environment variables with bash's native parameter expansion

doasu committed:
lock.sh :: optimised environment variables

replaced multiple environment variables with bash's native parameter expansion

doasu committed:
unlock.sh :: optimised environment variables

replaced multiple environment variables with bash's native parameter expansion

doasu committed:
Merge pull request #1 from doasu/doasu-patch-1

Optimised shell code

doasu committed:
main.tmux :: removed redundant commands

From my tests, the $(dirname) itself should suffice.

It does produce a string with two slashes, but that shouldn't be a problem.

doasu commented:
Hey there!
Just wanna say that this plugin is amazing and just what I was looking for.
I wanted to extend its functionality somehow -- log the lock state to a file (to present it in the status bar) -- and thought I might as well replace some environment variables with ones with parameter expansion ^^

Thanks for the plugin!
- doasu

doasu committed:
toggle.sh :: log lock state to file

doasu committed:
main.tmux :: corrected environment variables

missed some :p

doasu committed:
lock.sh :: log state to file

doasu committed:
unlock.sh :: log state to file

doasu committed:
Update lock.sh

TheSast reviewed (commented):
The variables changes look great, but as I mentioned previously, the log file used for the statusline should not have formatting hardcoded by the plugin.

If all's good I can revert the "log state to file" commits and squash merge this in, and then I can push a small change to make it easier for users to create their own format or script in general, with some examples in bash.

doasu committed:
toggle.sh :: removed duplicated writes to file

`lockstate` is already written to in both lock/unlock scripts. There's no need for it here.

TheSast committed:
revert: "unlock.sh :: log state to file"

This reverts commit b9bdcec352ac862bf8f0ae6492bc09e2bb2c9ad7.
The formatting should not be hardcoded.

TheSast committed:
revert: "lock.sh :: log state to file"

This reverts commit 3ab7f5055d44c4ed81289627646a2f781d937f59.
The formatting should not be hardcoded.

TheSast committed:
fix: turn relative script path absolute

TheSast commented:
@doasu I pushed up a [commit](https://github.com/TheSast/tmux-keylocker/commit/af1e46e8e8139da96163d3cd6375c5df939d0245) with that tmux option I mentioned, I also added the examples from my previous comment in the [README](https://github.com/TheSast/tmux-keylocker#configuration).
If you have any further questions feel free to ask here, or if you have any new ideas that you feel may warrant it, open up a new issue.

doasu commented:
Hey, just wanna say a huge thank you for implementing this!!
This is exactly what I had in mind \\^-^/
